### PR TITLE
Allow PCH builds with 'BUILD_SHARED_LIBS=1'

### PR DESCRIPTION
**Pull request by vittorioromeo**
Let's see if CI is happy...

### CODE REVIEW

**Comment by vittorioromeo** [comment]
SFMLBuildMaster: Build this please.

**Comment by eXpl0it3r** [comment]
> Let's see if CI is happy...

Did you intend to set `SFML_ENABLE_PCH` in the CI somewhere, because as is, this is not tested on the CI and thus won't make CI unhappy 😄

File: CMakeLists.txt [hunk 1]
```diff
@@ -211,6 +211,19 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER "CMake")
 
+# option to enable precompiled headers
+sfml_set_option(SFML_ENABLE_PCH FALSE BOOL "TRUE to enable precompiled headers for SFML builds -- only supported on Windows/Linux")
+
+if(SFML_ENABLE_PCH AND SFML_OS_MACOS)
+    message(FATAL_ERROR "Precompiled headers are currently not supported in macOS builds")
+endif()
```
Comment (eXpl0it3r) [inline CMakeLists.txt:219]: Can we leave the option (and error) at the original position as well?

Comment (vittorioromeo) [reply CMakeLists.txt:219]: Yes, will move it back.

**Pull request closed without merging**